presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java

```diff
@@ -125,6 +138,14 @@ public Block readBlock(ColumnDescriptor columnDescriptor, Type type)
         return columnReader.readBlock(type);
     }
 
+    private byte[] allocateBlock(int length)
+    {
+        byte[] buffer = new byte[length];
+        LocalMemoryContext blockMemoryContext = currentRowGroupMemoryContext.newLocalMemoryContext();
+        blockMemoryContext.setBytes(buffer.length);
```

Comment:
@dain any comments on this one? `OrcInputStream` is doing something similar with its `bufferMemoryUsage`.